archaius-core/src/main/java/com/netflix/archaius/ConfigListener.java

```diff
@@ -22,14 +22,28 @@
  * 
  * @author elandau
  */
-public interface DynamicConfigObserver {
+public interface ConfigListener {
+    /**
+     * Notification that a configuration was added.  This will normally only be called
+     * for CompositeConfig derived implementation
+     * @param config
+     */
+    public void onConfigAdded(Config config);
+    
+    /**
+     * Notification that a configuration was removed.  This will normally only be called
+     * for CompositeConfig derived implementation
+     * @param config
+     */
+    public void onConfigRemoved(Config config);
+    
     /**
      * Notify the parent that the value of a key has changed.  The parent is expected
      * to refresh its complete state.
      * 
      * @param propName
      */
-    public void onUpdate(String propName, Config config);
+    public void onConfigUpdated(String propName, Config config);
```

Comment:
If it's a snapshot update then on onConfigUpdated(Config config) will be called.  It it's an incremental update then onConfigUpdated(String propName, Config config) will be called multiple times but onConfigUpdated(Config config) will not be called.